internal/controller/controllers/preprovisioningimage_controller.go

```diff
@@ -356,35 +356,38 @@ func (r *PreprovisioningImageReconciler) mapInfraEnvPPI() func(a client.Object)
 		reconcileRequests := []reconcile.Request{}
 
 		for i := range images {
-			// Don't queue if the Image URL and InfraEnv's URL is the same.
-			if images[i].Status.ImageUrl != infraEnv.Status.ISODownloadURL {
-				reconcileRequests = append(reconcileRequests, reconcile.Request{
-					NamespacedName: types.NamespacedName{
-						Namespace: images[i].Namespace,
-						Name:      images[i].Name,
-					},
-				})
-			}
+			reconcileRequests = append(reconcileRequests, reconcile.Request{
+				NamespacedName: types.NamespacedName{
+					Namespace: images[i].Namespace,
+					Name:      images[i].Name,
+				},
+			})
 		}
 		return reconcileRequests
 	}
 	return mapInfraEnvPPI
 }
 
-func (r *PreprovisioningImageReconciler) AddIronicAgentToInfraEnv(ctx context.Context, log logrus.FieldLogger, infraEnv *aiv1beta1.InfraEnv) (ctrl.Result, error) {
-	// Retrieve infraenv from the database
+// AddIronicAgentToInfraEnv updates the infra-env in the database with the ironic agent ignition config if required
+// returns true when the infra-env was updated, false otherwise
+func (r *PreprovisioningImageReconciler) AddIronicAgentToInfraEnv(ctx context.Context, log logrus.FieldLogger, infraEnv *aiv1beta1.InfraEnv) (bool, error) {
 	key := types.NamespacedName{
 		Name:      infraEnv.Name,
 		Namespace: infraEnv.Namespace,
 	}
 	infraEnvInternal, err := r.Installer.GetInfraEnvByKubeKey(key)
 	if err != nil {
 		log.WithError(err).Error("failed to get corresponding infraEnv")
-		return ctrl.Result{}, err
+		return false, err
 	}
-	ironicAgentImage, err := r.getIronicAgentImageByRelease(ctx, log, infraEnvInternal)
-	if err != nil {
-		log.WithError(err).Warningf("Failed to get ironic agent image by release for infraEnv: %s", infraEnv.Name)
+	ironicAgentImage := infraEnv.Spec.IronicAgentImageOverride
+	if ironicAgentImage == "" {
+		ironicAgentImage, err = r.getIronicAgentImageByRelease(ctx, log, infraEnvInternal)
+		if err != nil {
+			log.WithError(err).Warningf("Failed to get ironic agent image by release for infraEnv: %s", infraEnv.Name)
```

Comment:
Should this return with the error here? I'm not sure why it would make sense for the function to keep going.

Reply:
There's a default value that we use as a fallback in the next block.

Here the logic goes:
- If there's an override use it
- No override, try to get it from the hub release image
- Hub release failed and no override, use hardcoded/env driven default

We're going to remove the last one in a follow-up, but since it will require us to communicate to the user that we can't find an image in some way it will require a bit more effort.

Reply:
Is there a code path here where the user provides a value, an error happens in `getIronicAgentImageByRelease`, and then the code silently ignores the user-supplied value and uses the default instead?

If not, you can resolve this thread.

Reply:
If the user provides an override we don't call `getIronicImageByRelease` so no, that's not something that can happen.